GameOfLifeApp/GameLogic.cs

```diff
@@ -0,0 +1,153 @@
+        /// <param name="currentColumn">Column of a current cell.</param>
+        /// <returns>Returns number of alive neighbours.</returns>
+        private int NeighboursCount(int currentRow, int currentColumn)
+        {
+            int count = 0;
+            for (int row = currentRow - 1; row <= currentRow + 1; row++)
+            {
+                for (int column = currentColumn - 1; column <= currentColumn + 1; column++)
+                {
+                    int actualRow;
+                    int actualColumn;
+                    if (row < 0)
+                    {
+                        actualRow = CurrentField.GetLength(0) - 1;
+                    }
+                    else if (row == CurrentField.GetLength(0))
+                    {
+                        actualRow = 0;
+                    }
+                    else
+                    {
+                        actualRow = row;
+                    }
+                    if (column < 0)
+                    {
+                        actualColumn = CurrentField.GetLength(1) - 1;
+                    }
+                    else if (column == CurrentField.GetLength(1))
+                    {
+                        actualColumn = 0;
+                    }
+                    else
+                    {
+                        actualColumn = column;
+                    }
+                    count += CurrentField[actualRow, actualColumn] ? 1 : 0;
+                }
+            }
+            count -= CurrentField[currentRow, currentColumn] ? 1 : 0;
+
+            return count;
+        }
+
+        /// <summary>
+        /// Gets new boolean values according to statements.
+        /// </summary>
+        /// <param name="neighboursCount">Number of alive cells around a given cell.</param>
+        /// <param name="isCurrentAlive">Given cell is alive. Value of isCurrentAlive: true.</param>
+        /// <returns>Returns the next value for a given cell.</returns>
+        private bool GetCellOffsprings(int neighboursCount, bool isCurrentAlive)
+        {
+            if (!isCurrentAlive && neighboursCount == 3)
+            {
+                return true;
+            }
+            else if (isCurrentAlive && neighboursCount == 2 || neighboursCount == 3)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// This method generates a new field and transfers data to a generated field.       
+        /// </summary>
+        public void GetNextGeneration()
+        {
+            bool[,] generatedField = new bool[CountOfRows, CountOfColumns];
+
+            for (int row = 0; row < CountOfRows; row++)
+            {
+                for (int column = 0; column < CountOfColumns; column++)
+                {
+                    int count = NeighboursCount(row, column);
+                    bool value = GetCellOffsprings(count, CurrentField[row, column]);
```

Comment:
better to change names for variables, as value should be updated on what value is it and the count would be good to know is it neighbor count or maybe the whole alive cell count, make it easier to read your code, and check the naming in all your classes.

Reply:
Changed: Count -> countAliveNeighbours